docs/howtos/DataPlane Generation - DPG/09tsp_client.mdx

```diff
@@ -0,0 +1,182 @@
+# Getting started with `tsp-client`
+
+`tsp-client` is a simple command line tool to facilitate generating client libraries from TypeSpec.
+
+## Installation
+
+```
+npm install -g @azure-tools/typespec-client-generator-cli
+```
+
+## Usage
+
+```
+tsp-client <command> [options]
+```
+
+## Commands
+
+Use one of the supported commands to get started generating clients from a TypeSpec project.
+This tool will default to using your current working directory to generate clients in and will
+use it to look for relevant configuration files. To specify a different output directory, use
+the `-o` or `--output-dir` option.
+
+### init
+
+Initialize the client library directory using a tspconfig.yaml. When running this command pass in a path to a local or remote tspconfig.yaml with the `-c` or `--tsp-config` flag.
+
+The `init` command generates a directory structure following the standard pattern used across Azure SDK language repositories, creates a [tsp-location.yaml](#tsp-locationyaml) file to control generation, and performs an initial generation of the client library. If you want to skip client library generation, then pass the `--skip-sync-and-generate` flag.
+
+> IMPORTANT: This command should be run from the root of the repository.
+
+Example:
+
+```
+azure-sdk-for-python> tsp-client init -c https://github.com/Azure/azure-rest-api-specs/blob/431eb865a581da2cd7b9e953ae52cb146f31c2a6/specification/contosowidgetmanager/Contoso.WidgetManager/tspconfig.yaml
+```
+
+### update
+
+Sync and generate client libraries from a TypeSpec project. The `update` command will look for a [tsp-location.yaml](#tsp-locationyaml) file in your current directory to sync a TypeSpec project and generate a client library.
+
+Example:
+
+```
+azure-sdk-for-python/sdk/contosowidgetmanager/azure-contoso-widgetmanager> tsp-client update
+```
+
+### sync
+
+Sync a TypeSpec project with the parameters specified in tsp-location.yaml.
```

Comment:
maybe you can say that this is the first step of `update`, so if you want to split `update` up you can call `sync` and or `generate`

Reply:
Done :)